### PR TITLE
Add custom handling for hub authentication requests

### PR DESCRIPTION
**Pull request by DominicSherman**
image

### CODE REVIEW

File: cardstack/src/hooks/transaction-confirmation/use-message.ts [hunk 1]
```diff
@@ -2,11 +2,11 @@ import { useRouteParams } from './use-route-params';
 
 export const useMessage = () => {
   const {
-    transactionDetails: {
-      payload: { params },
-    },
+    transactionDetails: { payload },
   } = useRouteParams();
 
+  const { params } = payload;
```
Comment (SparkleFaerieCoder) [inline cardstack/src/hooks/transaction-confirmation/use-message.ts]: Nice refactor 👍

File: cardstack/src/hooks/transaction-confirmation/use-message.ts [hunk 2]
```diff
@@ -2,11 +2,11 @@ import { useRouteParams } from './use-route-params';
...
+
   const message = params[1].message;
```
Comment (SparkleFaerieCoder) [inline cardstack/src/hooks/transaction-confirmation/use-message.ts]: Unless you would prefer that distinction, would it be possible to refactor this to reuse the same code between `use-message` and `use-primary-type `and  pass a  type param (`primaryType`, `message`) to grab the specified key? I think we could do something like  `params[1][type]`.

**Review by SparkleFaerieCoder — APPROVED** [review]
Nothing blocking, just one comment on duplicative code.

File: cardstack/src/hooks/transaction-confirmation/use-message.ts (hunk 2 above)
Comment (DominicSherman) [reply cardstack/src/hooks/transaction-confirmation/use-message.ts]: Sure, I think we could just return the whole params[1] object and then allow it to pull off whatever as needed 👍 I'll update

Comment (SparkleFaerieCoder) [reply cardstack/src/hooks/transaction-confirmation/use-message.ts]: @DominicSherman I like that even better 🎉

**Pull request merged**